### DISCUSSION TITLE
How to add a commit message template for DCO?

### DISCUSSION THREAD

**Discussion by marvin-hansen** [release-plz/release-plz#955] [Q&A] [answered]
Hi,

I have a bit of a corner case and didn't really wanted to open an issue for that.

Specifically,  my project has been accepted into the linux foundation and part of their project requirements is that every contributor must sign the[ Developer Certificate of Origin (DCO)](https://opensource.com/article/18/3/cla-vs-dco-whats-difference) which in practice is relatively easy by just adding the following to a commit message:

Signed-off-by: Your Name <your.name@email.com>

Git can sign automatically using the -s argument

git commit -s -m 'My commit message'

DCO was conceived back in 2004 almost certainly before the advent of build bots and GitHub actions...

Because DCO is required on my repo, release-plz PR's consistently "fail" the checks until I've manually set the DCO to pass.

This is particularly inconvenient because the manual overwrite can only be done on the desktop website meaning out-of-schedule releases when on the go isn't feasible since the DCO overwrite is not supported in the GH mobile app. Don't ask me why.

And here is my question:

How to add a commit message template for DCO?

I can think of this in two ways:

1) Somehow add an email to the release bot and add the  -s argument to get to sign commits. Is that even possible?
or
2) Just place a commit message template somewhere (where?) with a pre-signed comment where release-plz just appends its generated message.

Any help would be appreciated.

That said, release-plz is an amazing project and works so reliable behind the scenes that I cannot imagine anymore  running a project without it.

**Comment by marcoieni** [comment, marked as answer]
Signing commits sounds like a good default. dependabot does the same.
Implemented in https://github.com/MarcoIeni/release-plz/pull/956 👍 
Also thank you for you kind words :) In which project are you using release-plz?
